flake.nix

```diff
@@ -338,16 +318,22 @@
                 version
                 cargoArtifacts;
 
-              cargoExtraArgs = "${cargoBuildExtraArgs} ${extraBuildArgs} --package ${cargoPackage}";
+              # pyo3 needs a Python interpreter in the build environment
+              # https://pyo3.rs/v0.17.3/building_and_distribution#configuring-the-python-version
+              nativeBuildInputs = with pkgs; [ pkg-config python3 ];
+              # A git binary is needed for some of the tests
+              buildInputs = with pkgs; [ git ];
+
+              cargoExtraArgs = "${cargoBuildExtraArgs} ${extraBuildArgs} --workspace";
               CARGO_PROFILE = profile;
             } // extraArgs);
 
           # To build Nickel and its dependencies statically we use the musl
           # libc and clang with libc++ to build C and C++ dependencies. We
           # tried building with libstdc++ but without success.
-          buildStaticPackage = { pnameSuffix, cargoPackage, extraBuildArgs ? "", extraArgs ? { } }:
-            (buildPackage {
-              inherit pnameSuffix cargoPackage;
+          buildStaticWorkspace = { pnameSuffix, extraBuildArgs ? "", extraArgs ? { } }:
```

Comment:
It's a detail but the ability to decouple `pnameSuffix` from the original cargo package name had a small upside, which is that the derivations would exhibit more specific names (`nickel-lang-cli-pkg` for example, instead of just `nickel-lang-cli`). It's not very important, but was reason for sometimes having a `pnameSuffix` that doesn't match precisely the package name.

Reply:
I'm not sure I understand: we don't use cargo package names anymore, so why would we need to decouple them from suffices? We can still have derivations with more specific names (like `nickel-lang-pkg` vs `nickel-lang`)

Reply:
Sure, my point I guess is that as it is, this PR will show `nickel-lang-cli>` in the logs for both `nickel-lang-cli` and `nickel-lang-cli-pkg`, while it didn't use to be the case, is that correct? I agree that we don't really care about the `pnameSuffix` mechanism in particular.

Reply:
Good point. I think the latest version is using `pnameSuffix` correctly now.